docs/api.rst

```diff
@@ -257,6 +257,43 @@ This section documents events related to :class:`Client` and its connectivity to
     :param kwargs: The keyword arguments for the event that raised the
         exception.
 
+.. function:: on_gateway_error(event, data, shard_id, exc)
+
+    When a (known) gateway event cannot be parsed, a traceback is printed to
+    stderr and the exception is ignored by default. This should generally
+    not happen and is usually either a library issue, or caused by a breaking API change.
+
+    To change this behaviour, for example to completely stop the bot, this event can be overridden.
+
+    This can also be disabled completely by passing ``enable_gateway_error_handler=False``
+    to the client on initialization, restoring the pre-v2.6 behavior.
+
+    .. versionadded:: 2.6
+
+    .. note::
+        ``on_gateway_error`` will only be dispatched to :meth:`Client.event`.
+
+        It will not be received by :meth:`Client.wait_for`, or, if used,
+        :ref:`ext_commands_api_bot` listeners such as
+        :meth:`~ext.commands.Bot.listen` or :meth:`~ext.commands.Cog.listener`.
```

Comment:
why not?

Reply:
No particular reason, it keeps the behavior similar to `Client.on_error` (which has the same comment), and I'd argue using `wait_for` doesn't really make sense with this event anyway :p